### ISSUE TITLE
create github action that purges modified static files

### ISSUE THREAD

**Issue by tripodsan** [adobe/helix-pages#465]
similar to https://github.com/adobe/theblog/blob/master/.github/workflows/purge-code.yaml, just a bit smarter to automatically purge the files that were pushed.

**Comment by davidnuescheler** [comment]
this possibly should be a node implementation over shell script

**Comment by tripodsan** [comment]
see an example on how to react on a push event:
https://github.com/adobe-rnd/github-touch-action

**Comment by koraa** [comment]
Could you be more specific describing the goal of this ticket? E.g. user story style could help…what is the current behavior, how would we like that behavior to change; where should this action run? Serverless action…the script linked uses a github action which…seems like a bit of an odd choice…

**Comment by trieloff** [comment]
The Blog currently uses a project-specific GH action to purge the outer and inner CDN cache of all known static files when a push occurs. It would be easier if projects could simply refer to a pre-built action instead of each project building their own script. 

This pre-built action should use the same HLXPURGE method of purging, but it should figure out which files to purge based on the pushed commits and the base URL from a config. 

Building a GH action is easier than building a full-blown GH app.

**Comment by koraa** [comment]
So what would be the user flow to enable this? Shouldn't it be enabled by default?

**Comment by koraa** [comment]
Btw, the HLXPURGE HTTP verb, have we considered the potential impact such an http method can have on the possibility of amplified DOS attacks when unauthenticated?

**Comment by tripodsan** [comment]
> So what would be the user flow to enable this? Shouldn't it be enabled by default?

the user would add a `purge-action.yaml` to his repositorys `.github/workflows` directory.

**Comment by trieloff** [comment]
> Btw, the HLXPURGE HTTP verb, have we considered the potential impact such an http method can have on the possibility of amplified DOS attacks when unauthenticated?

We have, and decided to leave it open.

**Comment by koraa** [comment]
How should the action determine which files to purge? It should probably for instance pure X.html for a changed X.md file. Where should these rewriting rules come from?

**Comment by koraa** [comment]
https://github.com/koraa/hlx-gh-action-purge-updated-cache

**Comment by trieloff** [comment]
- [ ] license
- [ ] code of conduct
- [] contributing

**Comment by trieloff** [comment]
> Where should these rewriting rules come from?

No rewriting needed, the action works as is

**Comment by tripodsan** [comment]
@koraa looks good so far. what's missing is a way to filter the files that should trigger the purge. e.g. only the modified files below `scripts/**` should be purged.

**Comment by koraa** [comment]
Wasn't a requirement in the ticket. Please specify a way to provide these filters…

**Comment by trieloff** [comment]
The outer CDN URL should be optional, ref--repo--owner.hlx.page can be used.

**Comment by koraa** [comment]
Can someone please test this in the blog to see if it actually works before I move the repo and reset the git history?

**Comment by tripodsan** [comment]
> Wasn't a requirement in the ticket. Please specify a way to provide these filters…

hmm...right. I forgot to mention it. best would be using an input property:

```
on: push

jobs:
  ci_trigger:
    runs-on: ubuntu-latest
    name: Clear helix pages cache
    steps:
      - name: Trigger
        id: trigger
        uses: adobe/helix-pages-action-purge-cache@master
        with:
          repo-token: ${{ secrets.MY_GITHUB_TOKEN }}
          include: "scripts/**"
```

**Comment by tripodsan** [comment]
> Can someone please test this in the blog to see if it actually works before I move the repo and reset the git history?

please move it to `adobe-rnd`. you can continue development there.

**Comment by koraa** [comment]
> > Wasn't a requirement in the ticket. Please specify a way to provide these filters…
> 
> hmm...right. I forgot to mention it. best would be using an input property:

Precisely which grammar should this use. Looks like glob patterns, but those are not really standardized; your example looks like using the `globstar` bash extension. What about the ksh93 `extglob` extension with support for the klenee star? Are we using some standardized library for glob matching? Maybe we should enforce consistency there…

Given that this is not entirely trivial, I think it would be better to open a ticket in the repo with the full specification once it's been moved rather than hash this out as part of this ticket unless @davidnuescheler thinks this urgently needs to be added.

**Comment by koraa** [comment]
Looks like I have insufficient permissions to move into adobe-rnd. Somebody else will have to go for it.

**Comment by tripodsan** [comment]
@koraa use https://www.npmjs.com/package/glob, and whatever pattern they support. that's what we use elsewhere.

IMO it is important to add this - you can also hardcode the `scripts` directory restriction, if you don't want to add it.

**Comment by koraa** [comment]
Please move that requirment into it's own ticket after the repo has been moved.

**Comment by tripodsan** [comment]
@koraa you need to give me admin rights on that repository, otherwise I cannot move it.

**Comment by tripodsan** [comment]
I created the issue https://github.com/koraa/github-purge-cache-action/issues/2

**Comment by koraa** [comment]
Right, I added @trieloff @tripodsan and @davidnuescheler any one of you should be able to move :)
